src/cpp/ports.cpp

```diff
@@ -835,11 +840,10 @@ char *Ports::collapsePath(char *p)
 void Ports::refreshMagic()
 {
     delete impl;
-    impl = new Port_Matcher;
+    impl = new Port_Matcher(ports.size());
```

Comment:
I moved allocating `impl->enump` here, because I think it should never be dependent on `generate_minimal_hash`.